File: SMeatSocialNetwork/SMeat.MODELS/Models/Location.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace SMeat.MODELS.Models
{
    public class Location
    {
        public Location()
        {
            Id = Guid.NewGuid().ToString();
        }

        [Key]
        public string Id { get; set; }

        public string City { get; set; }

        public string Street { get; set; }

        public string Country { get; set; }

        //Navigation properties

        public virtual List<User> Users { get; set; }

        public virtual List<Workplace> Workplaces { get; set; }
    }
}

File: SMeatSocialNetwork/SMeat.WEB/src/app/_models/location.ts
export class Location {
  id: string;
  city: string;
  street: string;
  country: string;

  constructor(id?: string, city?: string, street?: string, country?: string) {
    this.id = id;
    this.city = city;
    this.street = street;
    this.country = country;
  }
}
